paasta_tools/apply_external_resources.py

```diff
@@ -0,0 +1,76 @@
+#!/opt/venvs/paasta-tools/bin/python
+import os
+import sys
+from filecmp import cmp
+from shutil import copy
+from subprocess import CalledProcessError
+from subprocess import run
+from traceback import print_exc
+
+APPLIED_DIRECTORY = ".applied"
+
+
+# This script expects the KUBECONFIG environment variable to be set correctly
+def main(puppet_resource_root: str) -> int:
+    exit_code = 0
+    applied_resource_root = os.path.join(puppet_resource_root, APPLIED_DIRECTORY)
+
+    # Loop through everything in the puppet resource path
+    for root, dirs, files in os.walk(puppet_resource_root):
+        # modifying the 'dirs' variable in-place will update the order that os.walk visits things
+        if APPLIED_DIRECTORY in dirs:
+            dirs.remove(APPLIED_DIRECTORY)
+        dirs.sort()  # Need to apply things in the correct order
+
+        # Check to see if there's a difference between what Puppet created and
+        # what's been previously applied
+        for filename in sorted([f for f in files if f.endswith(".yaml")]):
```

Comment:
Here we sort files inside a single directory, but full paths are still in arbitrary order.
We need to sort full paths for all files from all directories.

Reply:
see above.